portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -960,24 +965,35 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	 * stable ID of the profile, or with null if no applicable
 	 * profiles are in use.
 	 */
-	'getDefaultHeatmapProfile': function () {
-	    return this.getGeneticProfiles()
+	'getDefaultHeatmapProfile': makeCachedPromiseFunction(
+		function(self, fetch_promise) {
+	    self.getGeneticProfiles()
 	    .then(function (gp_metadata_list) {
+		var mrna_profile_id = null;
+		var prot_profile_id = null;
 		for (var i = 0; i < gp_metadata_list.length; i++) {
 		    if (gp_metadata_list[i].genetic_alteration_type === "MRNA_EXPRESSION" &&
 			    gp_metadata_list[i].datatype === "Z-SCORE") {
-			return gp_metadata_list[i].id;
+			mrna_profile_id = gp_metadata_list[i].id;
+			break;
 		    }
 		}
 		for (var i = 0; i < gp_metadata_list.length; i++) {
 		    if (gp_metadata_list[i].genetic_alteration_type === "PROTEIN_LEVEL" &&
 			    gp_metadata_list[i].datatype === "Z-SCORE") {
-			return gp_metadata_list[i].id;
+			prot_profile_id = gp_metadata_list[i].id;
+			break;
 		    }
 		}
-		return null;
-	    }).promise();
-	},
+		if (mrna_profile_id !== null) {
+		    fetch_promise.resolve(mrna_profile_id);
+		} else if (prot_profile_id !== null) {
+		    fetch_promise.resolve(prot_profile_id);
+		} else {
+		    fetch_promise.reject();
```

Comment:
This is an error: the function is documented to resolve with `null` if data can be retrieved but is not applicable, that is what the functions that call it expect. If this function does not resolve, rendering of the Oncoprint will not continue.